### ISSUE TITLE
Add durability/shelf life to recipe

### ISSUE THREAD

**Issue by alexander-schranz** [schemaorg/schemaorg#3205]
Not sure about the correct english word. But would be nice to have beside cooktimes and preptimes also how long a created food can be eaten. So I can search for food which last atleast for a few days or something like that.

https://schema.org/Recipe

**Comment by whitten** [comment]
Could the shelf life of a recipe be calculated from the ingredients of the recipe or the preparation methods ?

Pickled things should have a longer shelf life than cooked things with the same ingredients. Freeze dried should last longer than things sitting on a shelf in a root cellar, or even sealed/canned food not exposed to the air.

**Comment by alexander-schranz** [comment]
@whitten dont think so that you can map from ingredient shelf life to the finished meal. Baking, cooking, changes things and so it has another sheld life.

**Comment by whitten** [comment]
I agree that you can’t just use the ingredient shelf life as the prepared food’s shelf life. 
I’m sure the preparationMethod would have a large impact as well. 

I’m sure there are many factors. Do you anticipate it being a single time range, or could a calculation be a useful appropriation?

**Comment by alexander-schranz** [comment]
I personally think a single time range would be enough and as said I would not calculate something think that isnt possible. The question is if there is any additional information required for storage of the food e.g. (fridge (storageTemperature), avoid sun, ...)

Additional there could also be fields in the future if a food can be "frozen" and how long it can stay "frozen". But that I would definitely see as additional other fields not related to this ones. But not an expert in that fields.

**Comment by MatthiasWiesmann** [comment]
Would it not make sense to add a resulting product field, and add the shelf-life to Product object? All products have a shelf life.

**Comment by alexander-schranz** [comment]
Who is deciding such things and can make process here?

**Comment by MatthiasWiesmann** [comment]
I think we need to do the following to make progress:

* Separate this bug into multiple feature requests (shelf-life, cooking instructions, etc). 
* Make sure the use-case is not covered by other vocabularies, like the gs1 Web Vocabulary 
   see for instance [gs1:consumerHandlingStorage](https://ref.gs1.org/voc/consumerHandlingStorage)
* Make a concrete proposal (Pull request).

**Comment by whitten** [comment]
So even a calculation of the shelf life of the result of following a recipe as the minimum of the shelf life of the ingredients would be better than no shelf life. 

Is the shelf life of bread made from stale flour different from the shelf life of stale bread ?

Is the shelf life of a result of following a recipe dependent on how the result is stored ?

Is there a different shelf life for an environment with room temperature and low humidity and the environment of freezing temperature and low humidity ?

Is there an idea of a property as an array of time ranges indexed by each of the different environments names ?
I can imagine storing the property as a JSON value with separate keys for each of the environment names. 

Is there already a list of environments with names and properties that we can name and then use as an index into the shelf life property ?

Looking to the title of this, durability is in some way tied to shelf life. 
Is durability a stand alone idea or is it durability for a purpose ?

I can imagine a scone’s durability as a paperweight is drastically different from its durability as a foodstuff
